jaeger-core/src/main/java/com/uber/jaeger/samplers/PerOperationSampler.java

```diff
@@ -57,36 +62,42 @@ public PerOperationSampler(int maxOperations, OperationSamplingParameters strate
 
   /**
    * Updates the probabilistic samplers for each operation
+   * @param strategies The parameters for operation sampling
    * @return true iff any samplers were updated
    */
-  public boolean update(OperationSamplingParameters strategies){
-    // Note that this isn't synchronized because ProbabilisticSampler#update and
-    // GuaranteedThroughputSampler#update are synchronized
-    boolean isUpdated;
+  public boolean update(OperationSamplingParameters strategies) {
+    boolean isUpdated = false;
 
     double lowerBound = strategies.getDefaultLowerBoundTracesPerSecond();
-    double defaultSamplingRate = strategies.getDefaultSamplingProbability();
+    ProbabilisticSampler defaultSampler = new ProbabilisticSampler(strategies.getDefaultSamplingProbability());
 
-    isUpdated = defaultSampler.update(defaultSamplingRate);
+    if (!defaultSampler.equals(this.defaultSampler)) {
+      this.defaultSampler = defaultSampler;
+      isUpdated = true;
+    }
 
     for (PerOperationSamplingParameters strategy : strategies.getPerOperationStrategies()) {
       String operation = strategy.getOperation();
       double samplingRate = strategy.getProbabilisticSampling().getSamplingRate();
-
-      GuaranteedThroughputSampler guaranteedThroughputSampler = operationNameToSampler.get(operation);
-
-      if (guaranteedThroughputSampler != null) {
-        isUpdated |= guaranteedThroughputSampler.update(samplingRate, lowerBound);
-      } else if (operationNameToSampler.size() < maxOperations) {
-        isUpdated = (operationNameToSampler
-                        .putIfAbsent(operation,
-                                     new GuaranteedThroughputSampler(samplingRate,lowerBound)))
-                        == null;
-      } else {
-        log.info("Exceeded the maximum number of operations({}) for per operations sampling",
-                 maxOperations);
+      GuaranteedThroughputSampler sampler = new GuaranteedThroughputSampler(samplingRate, lowerBound);
```

Comment:
the way the ratelimit is implemented is that when the sampler is created, it starts off with 0 credits which is incremented once a tick until it goes over 1. The issue is that the bucket is reset to 0 every minute and never has chance to grow so we'll end up with never emitting any lowerbound sampled traces. The default lowerbound rate limit is 1/10 minutes. I haven't checked the Java codebase but this is how it is in all other languages.